src/lib/Scenes/Home/Components/ForYou/index.tsx

```diff
@@ -37,58 +40,96 @@ export class ForYou extends React.Component<Props, State> {
   }
 
   componentDidMount() {
+    const { forYou } = this.props
     const rows: DataSourceRow[] = []
-    const artworkModules = this.props.forYou.artwork_modules || []
-    const artistModules = this.props.forYou.artist_modules && this.props.forYou.artist_modules.concat()
-    const fairsModule = this.props.forYou.fairs_module
-    rows.push({ type: "fairs", data: fairsModule })
-    for (let i = 0; i < artworkModules.length; i++) {
-      const artworkModule = artworkModules[i]
-      rows.push({ type: "artwork", data: artworkModule })
-      if ((i + 1) % 2 === 0) {
+    const artworkModules = forYou.artwork_modules || []
+    const artistModules = forYou.artist_modules && forYou.artist_modules.concat()
+    const fairsModule = forYou.fairs_module
+
+    rows.push({
+      type: "fairs",
+      data: fairsModule,
+    })
+
+    artworkModules.forEach((artworkModule, index) => {
+      rows.push({
+        type: "artwork",
+        data: artworkModule,
+      })
+
+      const alternateRow = (index + 1) % 2 === 0
+
+      if (alternateRow) {
         const artistModule = artistModules.shift()
         if (artistModule) {
-          rows.push({ type: "artist", data: artistModule })
+          rows.push({
+            type: "artist",
+            data: artistModule,
+          })
         }
       }
-    }
+    })
+
     this.setState({
       dataSource: this.state.dataSource.cloneWithRows(rows),
     })
   }
 
-  handleRefresh = () => {
-    this.setState({ isRefreshing: true })
-    const stopRefreshing = () => this.setState({ isRefreshing: false })
-
-    Promise.all(
-      this.state.modules.map(module => {
-        return new Promise((resolve, reject) => {
-          module.forceFetch(null, readyState => {
-            if (readyState.error) {
-              reject(readyState.error)
-            } else if (readyState.aborted) {
-              reject()
-            } else if (readyState.done) {
-              resolve()
-            }
+  handleRefresh = async () => {
+    this.setState({
+      isRefreshing: true,
+    })
+
+    try {
+      await Promise.all(
+        this.state.modules.map(module => {
+          return new Promise((resolve, reject) => {
+            const { props, state: { data, relayProp } } = module
+
+            relayProp.refetch({ ...props.rail, fetchContent: true }, null, error => {
+              if (error) {
+                reject({ error })
+              } else {
+                resolve()
+              }
+            })
```

Comment:
@l2succes - Not sure if old code was from old Relay but the `forceFetch` prop didn't exist on closer inspection; digging in the above seemed to work and fit with the pagination containers per the docs. Thoughts?

Reply:
`forceFetch` is definitely from the old relay API. The `modules` in question are actually refetch containers. I think it makes sense to leave them as is and not use pagination containers since we're not pagination the individual sections.

Reply:
Ah! Sorry I meant refetch